ChiModules/KEigenvalueSolver/IterativeOperations/set_k_source.cc

```diff
@@ -43,95 +46,130 @@ void KEigenvalue::Solver::
-                                 xs->nu_delayed_sigma_f[gprime] *
-                                 phi_prevp[gprime] / k_eff;
+
+            //============================== Within-groupset
+            if (apply_wgs_scatter_src) {
+              size_t  num_transfers =
+                  xs->transfer_matrices[ell].rowI_indices[g].size();
+
+              //============================== Loop over transfers
+              for (int t = 0; t < num_transfers; ++t) {
+                size_t gprime = xs->transfer_matrices[ell].rowI_indices[g][t];
+
+                if ((gprime >= gs_i) and (gprime <= gs_f)) {
+                  double sigma_sm = xs->transfer_matrices[ell].rowI_values[g][t];
+                  inscatter_g += sigma_sm * phi_oldp[gprime];
                 }
               }
             }
-          }
-          q_mom[g] += precursor_g;
+          }//if moment avail
+
+          q_mom[g] += inscatter_g;
+
+          //======================================== Apply fission
+          if (xs->is_fissile and (ell == 0)) {
+            double fission_g = 0.0;
+            //============================== Across-groupset
+            if (apply_ags_fission_src) {
+              //============================== Loop over groups
+              for (size_t gprime = first_grp; gprime <= last_grp; ++gprime) {
+                double nu_sig_f = (options.use_precursors) ?
+                    xs->nu_prompt_sigma_f[gprime] : xs->nu_sigma_f[gprime];
+
+                if ((gprime < gs_i) or (gprime > gs_f)) {
+                  fission_g += xs->chi[g] * nu_sig_f *
+                               phi_prevp[gprime] / k_eff;
+
+                  //============================== Delayed contributions
+                  if (options.use_precursors and xs->num_precursors > 0) {
+                    //============================== Loop over precursors
+                    for (int j = 0; j < xs->num_precursors; ++j) {
+                      fission_g += xs->chi_delayed[g][j] *
+                                   xs->precursor_yield[j] *
+                                   xs->nu_delayed_sigma_f[gprime] *
+                                   phi_prevp[gprime] / k_eff;
+                    }
+                  }//if use precursors and has precursors
+                }//if across groupset
+              }//for gprime
+            }//if across-groupset
+
+            //============================== Across-groupset
+            if (apply_wgs_fission_src) {
+              //============================== Loop over groups
+              for (size_t gprime = first_grp; gprime <= last_grp; ++gprime) {
+                double nu_sig_f = (options.use_precursors) ?
+                                  xs->nu_prompt_sigma_f[gprime] : xs->nu_sigma_f[gprime];
+
+                if ((gprime >= gs_i) and (gprime <= gs_f)) {
+                  fission_g += xs->chi[g] * nu_sig_f *
+                               phi_prevp[gprime] / k_eff;
```

Comment:
what is the last p for in "phi_prevp" ??

Reply:
This is the syntax used in the SetSource routine within the LinearBoltzmann::Solver class, I simply mirrored that here. What the last "p" is for is best answered by @Naktakala .